src/main/java/org/candlepin/subscriptions/jmx/TallyJmxBean.java

```diff
@@ -138,11 +139,33 @@ public void tallyAllAccountsByHourly(String start, String end) throws IllegalArg
       description = "Offset to start from (may be null to start from beginning)")
   @ManagedOperationParameter(name = "batchSize", description = "Batch size")
   public void migrateHardwareMeasurements(String snapshotId, int batchSize) {
+    if (batchSize == 0) {
+      log.info("Batch size cannot be zero");
```

Comment:
You need to throw an exception otherwise it will be non-obvious to the user that nothing happened.  Also might as well make the check `if (batchSize < 1)` since there are crazy people with their negative numbers out there.

Reply:
IllegalArgumentException should be fine

Reply:
Will do